### PR TITLE
Add Docker image release

### LINKED ISSUE #51: Published docker image?
I'm interested in experimenting with this on a GKE cluster, mainly to see if I can easily export the metrics via prometheus and link it to a horizontal pod autoscaling setup.

I could build my own docker image, but I'm a pretty lazy person. Is there are chance we could get an official image published to docker hub (`buildkite/agent-metrics` or similar?). The `buildkite/agent` image has been perfect for use in GKE, and this would be a good companion image.

### PR DESCRIPTION
**Pull request by DrJosh9000**
Also run the release pipeline in a separate BK pipeline.

Fixes #51

### CODE REVIEW

File: .buildkite/pipeline.release.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ steps:
     concurrency_group: "release_buildkite_metrics_github"
     plugins:
       - aws-assume-role-with-web-identity:
-          role-arn: arn:aws:iam::032379705303:role/pipeline-buildkite-buildkite-agent-metrics
+          role-arn: arn:aws:iam::172840064832:role/pipeline-buildkite-buildkite-agent-metrics-release
```
Comment (yob) [inline .buildkite/pipeline.release.yml:15]:
```suggestion
          role-arn: arn:aws:iam::032379705303:role/pipeline-buildkite-buildkite-agent-metrics```

Comment (yob) [inline .buildkite/pipeline.release.yml:15]:
```suggestion
          role-arn: arn:aws:iam::032379705303:role/pipeline-buildkite-buildkite-agent-metrics-release
```

Comment (yob) [inline .buildkite/pipeline.release.yml:15]:
```suggestion
          role-arn: arn:aws:iam::032379705303:role/pipeline-buildkite-buildkite-agent-metrics
```

File: .buildkite/pipeline.release.yml [hunk 2]
```diff
@@ -22,7 +22,7 @@ steps:
       queue: "elastic-runners"
     plugins:
       - aws-assume-role-with-web-identity:
-          role-arn: arn:aws:iam::445615400570:role/pipeline-buildkite-buildkite-agent-metrics
+          role-arn: arn:aws:iam::172840064832:role/pipeline-buildkite-buildkite-agent-metrics-release
```
Comment (yob) [inline .buildkite/pipeline.release.yml:25]:
```suggestion
          role-arn: arn:aws:iam::445615400570:role/pipeline-buildkite-buildkite-agent-metrics
```

Comment (yob) [inline .buildkite/pipeline.release.yml:25]:
```suggestion
          role-arn: arn:aws:iam::445615400570:role/pipeline-buildkite-buildkite-agent-metrics-release
```

File: .buildkite/pipeline.release.yml [hunk 3]
```diff
@@ -22,7 +22,7 @@ steps:
       queue: "elastic-runners"
     plugins:
       - aws-assume-role-with-web-identity:
-          role-arn: arn:aws:iam::445615400570:role/pipeline-buildkite-buildkite-agent-metrics
+          role-arn: arn:aws:iam::445615400570:role/pipeline-buildkite-buildkite-agent-metrics-release
       - aws-ssm#v1.0.0:
           parameters:
             GITHUB_RELEASE_ACCESS_TOKEN: /pipelines/buildkite/buildkite-agent-metrics/GITHUB_RELEASE_ACCESS_TOKEN
```
Comment (yob) [inline .buildkite/pipeline.release.yml:28]:
```suggestion
            GITHUB_RELEASE_ACCESS_TOKEN: /pipelines/buildkite/buildkite-agent-metrics-release/GITHUB_RELEASE_ACCESS_TOKEN
```

**Review by yob — APPROVED** [review]
LGTM 👍

**Comment by DrJosh9000** [comment]
There should be a trigger step in `.buildkite/pipeline.yml`...

**Pull request merged**